shoop/front/template_helpers/general.py

```diff
@@ -16,41 +16,50 @@
 from shoop.utils.translation import cache_translations_for_tree
 
 
-def _get_list_products(request):
-    return Product.objects.list_visible(
+def _get_list_products(context, n_products, ordering=None, filter_dict=None, orderable_only=True):
+    request = context["request"]
+    customer = request.customer
+    products_qs = Product.objects.list_visible(
         shop=request.shop,
-        customer=request.customer,
+        customer=customer,
         language=get_language(),
     )
 
+    if filter_dict:
+        products_qs = products_qs.filter(**filter_dict)
+    if ordering:
+        products_qs = products_qs.order_by(ordering)
+    if orderable_only:
+        products = []
+        for product in products_qs.all():
```

Comment:
This should maybe use `.iterator()`... otherwise you might be loading all list-visible products into Django memory, just to return a few of them.